RCTAppleHealthKit/RCTAppleHealthKit+Methods_Hearing.m

```diff
@@ -0,0 +1,83 @@
+//
+//  RCTAppleHealthKit+Methods_Hearing.m
+//  RCTAppleHealthKit
+//
+//  This source code is licensed under the MIT-style license found in the
+//  LICENSE file in the root directory of this source tree.
+//
+
+#import "RCTAppleHealthKit+Methods_Hearing.h"
+#import "RCTAppleHealthKit+Utils.h"
+#import "RCTAppleHealthKit+Queries.h"
+
+@implementation RCTAppleHealthKit (Methods_Hearing)
+
+- (void)hearing_getEnvironmentalAudioExposure:(NSDictionary *)input callback:(RCTResponseSenderBlock)callback
+ {
+     if (@available(iOS 13.0, *)) {
+         HKQuantityType *environmentalAudioExposureType = [HKQuantityType quantityTypeForIdentifier:HKQuantityTypeIdentifierEnvironmentalAudioExposure];
+         HKUnit *unit = [HKUnit decibelAWeightedSoundPressureLevelUnit];
+         NSUInteger limit = [RCTAppleHealthKit uintFromOptions:input key:@"limit" withDefault:HKObjectQueryNoLimit];
+         BOOL ascending = [RCTAppleHealthKit boolFromOptions:input key:@"ascending" withDefault:false];
+         NSDate *startDate = [RCTAppleHealthKit dateFromOptions:input key:@"startDate" withDefault:nil];
+         NSDate *endDate = [RCTAppleHealthKit dateFromOptions:input key:@"endDate" withDefault:[NSDate date]];
+         if(startDate == nil){
+             callback(@[RCTMakeError(@"startDate is required in options", nil, nil)]);
+             return;
+         }
+         NSPredicate * predicate = [RCTAppleHealthKit predicateForSamplesBetweenDates:startDate endDate:endDate];
+
+         [self fetchQuantitySamplesOfType:environmentalAudioExposureType
+                                     unit:unit
+                                predicate:predicate
+                                ascending:ascending
+                                    limit:limit
+                               completion:^(NSArray *results, NSError *error) {
+             if(results){
+                 callback(@[[NSNull null], results]);
+                 return;
+             } else {
+                 callback(@[RCTJSErrorFromNSError(error)]);
+                 return;
+             }
```

Comment:
This whole chunk of code is duplicated many times in this library (including in this file). A near future enhancement could be to add a utility function to parse the inputs and fetch the samples for a given type.

Reply:
Agreed, we should get a better approach with the migration to Swift! Glad if you can help us with that.